### PR TITLE
Add ignore_error_codes config option closes #431

### LINKED ISSUE #431: Add ignore_error_codes config option
**Is your feature request related to a problem? Please describe.**
Add ignore_error_codes config option to the GCP plugin across all tables etc, similar to the one we have for AWS. This can be used to cover the 403 / API disabled case.

**Describe the solution you'd like**
Having `ignore_error_codes` in the .spc file will automatically give the users the ability to ignore the error codes as per choice.

**Describe alternatives you've considered**
Currently, we have a handling of `403` API disabled across all the tables to handle the API disabled error codes.

**Additional context**

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
N/A
```
</details>

# Example query results
<details>
  <summary>Results</summary>

1. ~/.steampipe/config/gcp.spc  (**without** `ignore_error_codes` configuration)
2. `sqladmin.googleapis.com` API disabled
```
connection "gcp" {
  plugin    = "gcp"
  project = "YOUR_PROJECT_NAME"
  credentials = "~/.config/gcloud/application_default_credentials.json"
}
```

Query result for table `gcp_sql_database`

```
> select * from gcp_sql_database
Error: googleapi: Error 403: Cloud SQL Admin API has not been used in project 123456785634 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/sqladmin.googleapis.com/overview?project=123456785634 then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.
Details:
[
  {
    "@type": "type.googleapis.com/google.rpc.Help",
    "links": [
      {
        "description": "Google developers console API activation",
        "url": "https://console.developers.google.com/apis/api/sqladmin.googleapis.com/overview?project=123456785634"
      }
    ]
  },
  {
    "@type": "type.googleapis.com/google.rpc.ErrorInfo",
    "domain": "googleapis.com",
    "metadata": {
      "consumer": "projects/123456785634",
      "service": "sqladmin.googleapis.com"
    },
    "reason": "SERVICE_DISABLED"
  }
]
, accessNotConfigured (SQLSTATE HV000)
```
#
1. ~/.steampipe/config/gcp.spc  (**with** `ignore_error_codes` configuration)
2. `sqladmin.googleapis.com` API disabled
```
connection "gcp" {
  plugin    = "gcp"
  project = "YOUR_PROJECT_NAME"
  credentials = "~/.config/gcloud/application_default_credentials.json"
  ignore_error_codes = ["403"]
}
```

Query result for table `gcp_sql_database`

```
> select * from gcp_sql_database
+------+---------------+------+---------+-----------+-----------+-----------------------------------------+------------------------------------+-------+------+----------+---------+------+
| name | instance_name | kind | charset | collation | self_link | sql_server_database_compatibility_level | sql_server_database_recovery_model | title | akas | location | project | _ctx |
+------+---------------+------+---------+-----------+-----------+-----------------------------------------+------------------------------------+-------+------+----------+---------+------+
+------+---------------+------+---------+-----------+-----------+-----------------------------------------+------------------------------------+-------+------+----------+---------+------+

```

</details>

### CODE REVIEW

File: config/gcp.spc [hunk 1]
```diff
@@ -19,4 +19,8 @@ connection "gcp" {
   # `impersonate_service_account` (optional) - The GCP service account (string) which should be impersonated.
   # If not set, no impersonation is done.
   #impersonate_service_account = "YOUR_SERVICE_ACCOUNT"
+
+  # List of additional GCP error codes to ignore for all queries.
+  # By default, common not found error codes are ignored and will still be ignored even if this argument is not set.
+  #ignore_error_codes = ["AccessDenied", "AccessDeniedException", "NotAuthorized", "UnauthorizedOperation", "AuthorizationError", "403"]
```
Comment (misraved) [inline config/gcp.spc:25]: @ParthaI Are all the above error codes returned by the GCP API?

File: gcp/plugin.go [hunk 2]
```diff
@@ -20,6 +20,9 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 	p := &plugin.Plugin{
 		Name:             pluginName,
 		DefaultTransform: transform.FromCamel(),
+		DefaultGetConfig: &plugin.GetConfig{
+			ShouldIgnoreError: isIgnorableError([]string{"404", "400"}),
```
Comment (misraved) [inline gcp/plugin.go:24]: What does `404` and `400` error code stand for?

Comment (ParthaI) [reply gcp/plugin.go:24]: `404` is the not found error code and `400` is server error.

File: config/gcp.spc (hunk 1 above)
Comment (ParthaI) [reply config/gcp.spc:25]: No, I have just given the examples, should we update it as per API error?

Comment (misraved) [reply config/gcp.spc:25]: @ParthaI we should update it as per the API error.

**Comment by misraved** [comment]
@ParthaI could you please update the `docs/index.md` file to include information on `ignore_error_codes`? Please refer https://hub.steampipe.io/plugins/turbot/aws#configuration for more information.

**Comment by misraved** [comment]
@ParthaI do we still need the following code - https://github.com/turbot/steampipe-plugin-gcp/blob/3862653742bd4d8bb32b2e4b6af54d11d38edb4f/gcp/ignore_error_predicate.go#L18-L22

**Comment by misraved** [comment]
@ParthaI please take a look at https://cloud.google.com/resource-manager/docs/core_errors#:~:text=BAD_REQUEST%20(400),-Error%20code&text=The%20API%20request%20is%20invalid,could%20not%20understand%20the%20request.&text=The%20binary%20domain%20request%20is%20invalid. for more information on error codes.

**Review by misraved — APPROVED** [review]

**Pull request merged**